File: express-backend/src/services/provider/SwitchGameSearchProvider.ts
import { ExternalGameSearchResult, ProviderSearchResult } from '../../dto/external-search';
import { GameSearchProvider } from '../provider/GameSearchProvider';

// Switch 游戏搜索 Provider — 占位，与 Java 端 SwitchGameSearchProvider 对齐
export class SwitchGameSearchProvider implements GameSearchProvider {
  id(): string {
    return 'switch';
  }

  async search(query: string, page: number): Promise<ProviderSearchResult<ExternalGameSearchResult>> {
    return {
      provider: this.id(),
      enabled: false,
      message: 'Switch 暂无公开 API，保留占位',
      page,
      totalPages: 0,
      totalResults: 0,
      results: [],
    };
  }
}

File: legacy/java-backend/src/main/java/com/pixelreel/service/provider/SwitchGameSearchProvider.java
package com.pixelreel.service.provider;

import com.pixelreel.config.SwitchProperties;
import com.pixelreel.dto.external.ExternalGameSearchResult;
import com.pixelreel.dto.external.ProviderSearchResult;
import com.pixelreel.service.search.GameSearchProvider;
import java.util.Collections;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class SwitchGameSearchProvider implements GameSearchProvider {
  private final SwitchProperties switchProperties;

  @Override
  public String id() {
    return "switch";
  }

  @Override
  public ProviderSearchResult<ExternalGameSearchResult> search(String query, int page) {
    ProviderSearchResult<ExternalGameSearchResult> result = new ProviderSearchResult<>();
    result.setProvider(id());

    result.setEnabled(false);
    result.setMessage("Switch 暂无公开 API，保留占位");
    result.setPage(page);
    result.setTotalPages(0);
    result.setTotalResults(0);
    result.setResults(Collections.emptyList());
    return result;
  }
}

